src/plugins/api-proxy/util/send.js

```diff
@@ -0,0 +1,331 @@
+	endpoint,
+	params,
+	flags,
+	meta = {},
+}) => {
+	const dataParams = querystring.stringify(params);
+	const headers = { ...externalRequestOpts.headers };
+	let url = encodeURI(`/${endpoint}`);
+	let body;
+	const jar = createCookieJar(url);
+
+	if (flags || meta.flags) {
+		headers['X-Meetup-Request-Flags'] = (flags || meta.flags).join(',');
+	}
+
+	if (meta.metaRequestHeaders) {
+		headers[API_META_HEADER] = meta.metaRequestHeaders.join(',');
+	}
+
+	if (meta.variants) {
+		headers['X-Meetup-Variants'] = Object.keys(
+			meta.variants
+		).reduce((header, experiment) => {
+			const context = meta.variants[experiment];
+			const contexts = context instanceof Array ? context : [context];
+			header += contexts.map(c => `${experiment}=${c}`).join(' ');
+			return header;
+		}, '');
+	}
+
+	switch (externalRequestOpts.method) {
+		case 'patch':
+		case 'post':
+			if (externalRequestOpts.formData) {
+				break;
+			}
+			body = dataParams;
+			headers['content-type'] = 'application/x-www-form-urlencoded';
+			break;
+		case 'delete':
+		case 'get':
+		default:
+			url += `?${dataParams}`;
+			headers['content-type'] = 'application/json';
+			headers['X-Meta-Photo-Host'] = 'secure';
+	}
+
+	const externalRequestOptsQuery = {
+		...externalRequestOpts,
+		headers,
+		jar,
+		url,
+	};
+
+	// only add body if defined
+	if (body) {
+		externalRequestOptsQuery.body = body;
+	}
+
+	return externalRequestOptsQuery;
+};
+
+export function getAuthHeaders(request) {
+	const oauth_token =
+		request.state.oauth_token || request.plugins.requestAuth.oauth_token; // browser-based requests // internal server requests
+	if (!request.state.MEETUP_MEMBER && oauth_token) {
+		return {
+			authorization: `Bearer ${oauth_token}`,
+		};
+	}
+	const cookies = { ...request.state };
+	const csrf = uuid.v4();
+	cookies.MEETUP_CSRF = csrf;
+	cookies.MEETUP_CSRF_DEV = csrf;
+	const cookie = Object.keys(cookies)
+		.map(name => `${name}=${cookies[name]}`)
+		.join('; ');
+
+	return {
+		cookie,
+		'csrf-token': csrf,
+	};
+}
+
+export function getLanguageHeader(request) {
+	const requestLang = request.getLanguage();
+	const headerLang = request.headers['accept-language'];
+	const acceptLang =
+		requestLang && headerLang
+			? `${requestLang},${headerLang}`
+			: requestLang || headerLang;
+	return acceptLang;
+}
+
+export function parseRequestHeaders(request) {
+	const externalRequestHeaders = {
+		...request.headers,
+		...getAuthHeaders(request),
+		'accept-language': getLanguageHeader(request),
+	};
+
+	delete externalRequestHeaders['host']; // let app server set 'host'
+	delete externalRequestHeaders['accept-encoding']; // let app server set 'accept'
+	delete externalRequestHeaders['content-length']; // original request content-length is irrelevant
+	delete externalRequestHeaders['content-type']; // the content type will be set in buildRequestArgs
+
+	// cloudflare headers we don't want to pass on
+	delete externalRequestHeaders['cf-ray'];
```

Comment:
We should be able to get rid of these now, since we aren't behind cloudflare anymore.